FiveCalls/FiveCalls/LocationSheet.swift

```diff
@@ -13,7 +13,7 @@ struct LocationSheet: View {
     @Environment(\.dismiss) var dismiss
     
     let location: NewUserLocation?
-    let delegate: AppStateDelegate?
+    let setLocation: @MainActor (NewUserLocation) -> () // really? you have to specify mainactor here too?
```

Comment:
I too admit I don't entirely fully grok `MainActor` yet, even after reading up on it a bit more. It sounds magical, like you just apply it and everything within scope runs on the main queue.

Reply:
I read a bit more about actors in general, which I think I get, but I am still baffled by how you should structure `@MainActor` so you don't end up with even *more* main-handling code.